user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotas.java

```diff
@@ -0,0 +1,201 @@
+
+    /**
+     * Determine whether the given user has quotas.
+     *
+     * @param username Name of the user
+     *
+     * @return True if the user exists
+     */
+    public boolean exists(String username) {
+        byte[] data = zkClient.readData("/config/users/" + username, true);
+
+        if (data != null)   {
+            String jsonString = new String(data, Charset.defaultCharset());
+            JsonObject json = new JsonObject(jsonString);
+            validateJsonVersion(json);
+            JsonObject config = json.getJsonObject("config");
+
+            if (config != null) {
+                String prod = config.getString("producer_byte_rate");
+                String cons = config.getString("consumer_byte_rate");
+                String perc = config.getString("request_percentage");
+
+                if (prod != null || cons != null || perc != null) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Delete the quotas for the given user.
+     * It is not an error if the user doesn't exist, or doesn't currently have any quotas.
+     *
+     * @param username Name of the user
+     */
+    public void delete(String username) {
+        byte[] data = zkClient.readData("/config/users/" + username, true);
+
+        if (data != null)   {
+            log.debug("Deleting quotas for user {}", username);
+            zkClient.writeData("/config/users/" + username, deleteUserJson(data));
+            notifyChanges(username);
+        } else {
+            log.warn("Quotas for user {} already don't exist", username);
+        }
+    }
+
+    /**
+     * Deletes the quotas from existing JSON
+     *
+     * @param user JSON string with existing user configuration as byte[]
+     *
+     * @return  Returns the updated JSON without the quotas as byte array
+     */
+    protected byte[] deleteUserJson(byte[] user)   {
+        JsonObject json = new JsonObject(new String(user, Charset.defaultCharset()));
+
+        validateJsonVersion(json);
+
+        if (json.getJsonObject("config") == null)   {
+            json.put("config", new JsonObject());
+        }
+
+        if (json.getJsonObject("config").getString("producer_byte_rate") != null) {
+            json.getJsonObject("config").remove("producer_byte_rate");
+        }
+        if (json.getJsonObject("config").getString("consumer_byte_rate") != null) {
+            json.getJsonObject("config").remove("consumer_byte_rate");
+        }
+        if (json.getJsonObject("config").getString("request_percentage") != null) {
+            json.getJsonObject("config").remove("request_percentage");
+        }
+
+        return json.encode().getBytes(Charset.defaultCharset());
+    }
+
+    protected void validateJsonVersion(JsonObject json) {
+        if (json.getInteger("version") != 1) {
+            throw new RuntimeException("Failed to validate the user JSON. The version is missing or has an invalid value.");
```

Comment:
Doing this is fine only if we have tests that would detect if a new version of Kafka started creating users with a different version number.

Reply:
@stanlyDoge do we have tests as described above?

Reply:
No, I do not think so. Is it the same case for `ScramShaCredentials`? https://github.com/strimzi/strimzi-kafka-operator/blob/master/user-operator/src/main/java/io/strimzi/operator/user/operator/ScramShaCredentials.java#L230